### PR TITLE
add chpter3 linear-networks of paddle

### PR DESCRIPTION
**Pull request by Shelly111111**
contribute chpter3 linear-networks of paddle

### CODE REVIEW

File: chapter_linear-networks/image-classification-dataset.md [hunk 1]
```diff
@@ -139,6 +158,27 @@ def show_images(imgs, num_rows, num_cols, titles=None, scale=1.5):  #@save
     return axes
 ```
 
+```{.python .input}
+#@tab paddle
+def show_images(imgs, num_rows, num_cols, titles=None, scale=1.5):  #@save
```
Comment (cheungdaven) [inline chapter_linear-networks/image-classification-dataset.md:163]: move #@save to the line above, like:
#@save
def show_images(...)

File: chapter_linear-networks/linear-regression-concise.md [hunk 2]
```diff
@@ -77,6 +84,14 @@ def load_array(data_arrays, batch_size, is_train=True):  #@save
     return dataset
 ```
 
+```{.python .input}
+#@tab paddle
+def load_array(data_arrays, batch_size, is_train=True):  #@save
```
Comment (cheungdaven) [inline chapter_linear-networks/linear-regression-concise.md:89]: move #@save to the line above

File: chapter_linear-networks/linear-regression.md [hunk 3]
```diff
@@ -247,18 +247,35 @@ import numpy as np
...
+```
+
 为了说明矢量化为什么如此重要，我们考虑(**对向量相加的两种方法**)。
 我们实例化两个全为1的10000维向量。
 在一种方法中，我们将使用Python的for循环遍历向量；
 在另一种方法中，我们将依赖对`+`的调用。
 
 ```{.python .input}
-#@tab all
+#@tab mxnet, pytorch, tensorflow
```
Comment (cheungdaven) [inline chapter_linear-networks/linear-regression.md:266]: is it possible to reuse this code? e.g., we can redefine d2l.ones() in paddle.py.

File: chapter_linear-networks/softmax-regression-scratch.md [hunk 4]
```diff
@@ -245,6 +271,22 @@ def accuracy(y_hat, y):  #@save
     return float(d2l.reduce_sum(d2l.astype(cmp, y.dtype)))
 ```
 
+```{.python .input}
```
Comment (cheungdaven) [inline chapter_linear-networks/softmax-regression-scratch.md:274]: it would be better if we can reuse this function.

**Review by cheungdaven — CHANGES_REQUESTED** [review]

File: chapter_linear-networks/softmax-regression-scratch.md (hunk 4 above)
Comment (Shelly111111) [reply chapter_linear-networks/softmax-regression-scratch.md:274]: paddle.Tensor does not support 0-dimension operation, so it is easy to add a layer of dimension when using it. Here, the size must be judged once. If the size is wrong, it is likely to cause calculation errors.

Comment (Shelly111111) [reply chapter_linear-networks/softmax-regression-scratch.md:274]: paddle.Tensor does not support 0-dimension operation, so it is easy to add a layer of dimension when using it. Here, the size must be judged once. If the size is wrong, it is likely to cause calculation errors.

File: chapter_linear-networks/linear-regression-concise.md (hunk 2 above)
Comment (Shelly111111) [reply chapter_linear-networks/linear-regression-concise.md:89]: other #@save need move above？

**Review by AnirudhDagar — CHANGES_REQUESTED** [review]
d2l lib changes are not reflected in this PR. Please run `d2lbook build lib` and commit the changes. See [comment here](https://github.com/d2l-ai/d2l-zh/pull/1116#issuecomment-1099657466) You'll need to rebase this PR against the paddle branch before that. Thanks @Shelly111111 😄 

Also, it would be great to get these changes in fast so that we can get this merged and other PRs (#1116, #1122) are unblocked.
cc @cheungdaven

**Comment by tngt** [comment]
> d2l lib changes are not reflected in this PR. Please run `d2lbook build lib` and commit the changes. See [comment here](https://github.com/d2l-ai/d2l-zh/pull/1116#issuecomment-1099657466) You'll need to rebase this PR against the paddle branch before that. Thanks @Shelly111111 😄
> 
> Also, it would be great to get these changes in fast so that we can get this merged and other PRs (#1116, #1122) are unblocked. cc @cheungdaven

done,  please check #1131

**Pull request closed without merging**